sonar-scala-plugin/src/test/java/org/sonarsource/scala/externalreport/scalastyle/ScalastyleFamilySensorTest.java

```diff
@@ -0,0 +1,252 @@
+
+  @Test
+  public void no_issues_with_invalid_xml_report() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "invalid.xml");
+    assertThat(externalIssues).isEmpty();
+    String realPath = PROJECT_DIR.toRealPath().resolve("invalid.xml").toString();
+    assertThat(logTester.logs(LoggerLevel.ERROR)).containsExactlyInAnyOrder(
+      "No issues information will be saved as the report file '" + realPath + "' can't be read.");
+  }
+
+  @Test
+  public void issues_when_xml_file_has_errors() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "scalastyle-with-errors.xml");
+    assertThat(externalIssues).hasSize(2);
+
+    ExternalIssue first = externalIssues.get(0);
+    assertThat(first.primaryLocation().inputComponent().key()).isEqualTo("project:HelloWorld.scala");
+    assertThat(first.ruleKey().toString()).isEqualTo("scapegoat:EmptyCaseClass");
+    assertThat(first.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(first.severity()).isEqualTo(Severity.MINOR);
+    assertThat(first.primaryLocation().message()).isEqualTo("Valid issue");
+    assertThat(first.primaryLocation().textRange().start().line()).isEqualTo(5);
+
+    ExternalIssue second = externalIssues.get(1);
+    assertThat(second.primaryLocation().inputComponent().key()).isEqualTo("project:HelloWorld.scala");
+    assertThat(second.ruleKey().toString()).isEqualTo("scalastyle:unknown.key");
+    assertThat(second.type()).isEqualTo(RuleType.CODE_SMELL);
+    assertThat(second.severity()).isEqualTo(Severity.MAJOR);
+    assertThat(second.primaryLocation().message()).isEqualTo("Missing line");
+    assertThat(second.primaryLocation().textRange()).isNull();
+
+    assertThat(logTester.logs(LoggerLevel.ERROR)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactlyInAnyOrder(
+      "Fail to resolve 1 file(s). No issues imported related to file(s): /absolute/path/to/InvalidPath.scala");
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactlyInAnyOrder(
+      "Missing information or unsupported file type for source:'', file:'/absolute/path/to/HelloWorld.scala', message:'Missing source'",
+      "Missing information or unsupported file type for source:'com.sksamuel.scapegoat.inspections.EmptyCaseClass', file:'/absolute/path/to/HelloWorld.scala', message:''",
+      "Missing information or unsupported file type for source:'', file:'/absolute/path/to/HelloWorld.scala', message:''");
+  }
+
+  @Test
+  public void invalid_line_number() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 2, "scalastyle-invalid-line.xml");
+    assertThat(externalIssues).isEmpty();
+    String realPath = PROJECT_DIR.toRealPath().resolve("scalastyle-invalid-line.xml").toString();
+    assertThat(logTester.logs(LoggerLevel.ERROR)).containsExactlyInAnyOrder(
+      "No issues information will be saved as the report file '" + realPath + "' can't be read.");
```

Comment:
The message is not really helpful here...

Reply:
I will add the underlying exception message